java/ql/lib/semmle/code/java/Annotation.qll

```diff
@@ -60,16 +131,78 @@ class Annotation extends @annotation, Expr {
 
   /**
    * Gets a value of the annotation element with the specified `name`, which must be declared as an array
-   * type.
+   * type. This includes default values in case no explicit value is specified.
+   *
+   * If the annotation element is defined with an array initializer, then the result will be one of the
+   * elements of that array. Otherwise, the result will be the single expression defined for the value.
```

Comment:
```suggestion
   * elements of that array. Otherwise, the result will be the array-typed expression passed for this value.
```
(and similarly all the other predicates duplicating this text)

Reply:
What do you mean by "array-typed"? For example when Java code uses `@Target(ElementType.ANNOTATION_TYPE)`, then `ElementType.ANNOTATION_TYPE` is a single expression for the `ElementType[] value` element. And this predicate will have that single `ElementType.ANNOTATION_TYPE` field read as result.

The "Otherwise, ..." sentence is supposed to cover this exact case where a single expression and no array initializer is used. Maybe I am misunderstanding your comment.

Reply:
Ah sure, I thought you wanted to address the case where an array is passed, rather than a value of the array's element type which is implicitly wrapped in an array. Ignore my suggestion.